### ISSUE TITLE
Review these latest docs for beta

### ISSUE THREAD

**Issue by led76marx** [thenewboston-developers/Website#1752]
Need to review these latest docs. Bucky has updated them for beta and also added some new diagrams.

- Components: https://thenewboston.com/guide/components
- Blocks: https://thenewboston.com/guide/blocks
- Flows: https://thenewboston.com/guide/flows
- Network Initialization: https://thenewboston.com/guide/network-initialization
- Scheduling: https://thenewboston.com/guide/scheduling

**Comment by buckyroberts** [comment]
This one is already ready.

- Schedule Adjustments: https://thenewboston.com/guide/schedule-adjustments

**Comment by led76marx** [comment]
@buckyroberts would you prefer me to copy them in a gdoc and comment there, or directly do an edit pass? Either would work for me. Plan to go through all of them tomorrow.

**Comment by dmugtasimov** [comment]
@buckyroberts it would be great to have some contex comment enabled space to work on the document as @led76marx suggested. Google Docs or Confluence would work. If there are other options, please, suggest.

**Comment by dmugtasimov** [comment]
@buckyroberts @led76marx https://docs.google.com/document/d/1EgS2xZTVOSG_YiNWjXV6a1FSXErdQFcyOhmCstjI22A/edit?usp=sharing - this link should have comment permission

**Comment by dmugtasimov** [comment]
@buckyroberts I added comments to the google doc. The link is in previous comment.

**Comment by dmugtasimov** [comment]
@buckyroberts @led76marx Link for blocks: https://docs.google.com/document/d/1nRXJ2X55J8nm_c9n8brGDJp2ay5h7q4JWznAPxLTU-Y/edit?usp=sharing

**Comment by dmugtasimov** [comment]
@buckyroberts added questions and comments to the prev link

**Comment by dmugtasimov** [comment]
@buckyroberts @led76marx Flows link: https://docs.google.com/document/d/1e8vimpUxoujvGYmsN6KYJwt-yluAJuRwlVkiUV3ilAE/edit?usp=sharing

**Comment by dmugtasimov** [comment]
@buckyroberts I suggest to where I have questions in the documents instead of just answering the question modify the document that way so next reader would ask the same question (this way we will not just answer my questions but also improve the document). I need your email to grant write permissions.

**Comment by dmugtasimov** [comment]
@buckyroberts asked some questions about Flows

**Comment by dmugtasimov** [comment]
Link for network initialization: https://docs.google.com/document/d/1EPxuU8tPPHwMDjTovqXm7m-Rw7NDhseXsWhbv45ZUx0/edit?usp=sharing

**Comment by dmugtasimov** [comment]
@buckyroberts 
I think we should have a better way to implement documentation. Where documentation is more separated from source code. Maybe to use some sort of template engine.

```
import React, {FC} from 'react';

import {DocContainer, DocImage, DocList} from 'components';

import GenesisBlock from './GenesisBlock.png';
import GenesisBlockComposition from './GenesisBlockComposition.png';

const GuideNetworkInitialization: FC = () => {
  return (
    <DocContainer className="GuideNetworkInitialization" title="Network Initialization" lastUpdated="21 Apr 2021">
      <p>
        The network is initialized from a single node and a genesis block. This block is the very first block in the
        blockchain.
      </p>
      <DocImage alt="genesis block" maxWidth={400} src={GenesisBlock} />
      <p>
        The genesis block consists of the minimum amount of information required to initialize a new network including:
      </p>
      <DocList variant="ul">
        <li>All account information from alpha</li>
        <li>A record of the origin node</li>
        <li>The initial schedule consisting a single node (the origin node)</li>
      </DocList>
      <DocImage alt="genesis block composition" maxWidth={500} src={GenesisBlockComposition} />
    </DocContainer>
  );
};

export default GuideNetworkInitialization;
```

**Comment by dmugtasimov** [comment]
Scheduling: https://docs.google.com/document/d/1I6MGi1WD4P5ujPzTo5g62GFdMlj1XBRMbJMN-b-qvUo/edit?usp=sharing

**Comment by dmugtasimov** [comment]
@buckyroberts I did not actually review scheduling because I need to resolve the comments I have already made to move forward.

**Comment by led76marx** [comment]
> @buckyroberts
> I think we should have a better way to implement documentation. Where documentation is more separated from source code. Maybe to use some sort of template engine.
> 
> ```
> import React, {FC} from 'react';
> 
> import {DocContainer, DocImage, DocList} from 'components';
> 
> import GenesisBlock from './GenesisBlock.png';
> import GenesisBlockComposition from './GenesisBlockComposition.png';
> 
> const GuideNetworkInitialization: FC = () => {
>   return (
>     <DocContainer className="GuideNetworkInitialization" title="Network Initialization" lastUpdated="21 Apr 2021">
>       <p>
>         The network is initialized from a single node and a genesis block. This block is the very first block in the
>         blockchain.
>       </p>
>       <DocImage alt="genesis block" maxWidth={400} src={GenesisBlock} />
>       <p>
>         The genesis block consists of the minimum amount of information required to initialize a new network including:
>       </p>
>       <DocList variant="ul">
>         <li>All account information from alpha</li>
>         <li>A record of the origin node</li>
>         <li>The initial schedule consisting a single node (the origin node)</li>
>       </DocList>
>       <DocImage alt="genesis block composition" maxWidth={500} src={GenesisBlockComposition} />
>     </DocContainer>
>   );
> };
> 
> export default GuideNetworkInitialization;
> ```

I second that. Main reason being that we might also need a search functionality in the future.

**Comment by buckyroberts** [comment]
@dmugtasimov @led76marx @ma2b0043 

All of the guide and governor doc pages should be updated now based on your feedback:
- https://thenewboston.com/guide/introduction
- https://thenewboston.com/governance/overview

If you guys wouldn't mind going through and doing a deep dive on everything that would be great. A lot of work I know but I want to make sure that we are all on the same page regarding exactly what we are building before moving forward.

The original documentation was written assuming that there would only be one main type of block (for sending coins), however now we are storing more information on the blockchain such as:
- governor votes
- username registrations
- node boosting
- etc…

Because of this a lot of the terminology we used before no longer fits. For example, rather than "transfer" being used, which previously implied a transfer of coins, it now may cause confusion regarding exactly which objects are being transferred.

I went through and updated a lot of the terminology and data structures to support these changes, however if there are any improvements or optimizations that can be made to the architecture please feel free to suggest. We can discuss and then I can update the documentation and also the diagrams.

For some "todo" items:

The "root account file" logic is still out of date on the website. I didn't want to remove it yet to keep it for reference. That root account file is now the "genesis block", but will also include other information in addition to just account data. Also we still will need some kind of snapshot system to continuously backup the blockchain and also allow faster syncing / partial blockchains, but I figured this would be one of the last bits of logic we implement once everything else is in place so we can wait on that - would like @dmugtasimov 's feedback on the best way to implement that as well.

Another thing that we need to do is add all the fields and field types to each of the blocks here https://thenewboston.com/guide/block-types I am not sure how we want to display those but saw @dmugtasimov and @led76marx talking about a template engine which would be perfect for this. If you could get together and let @ma2b0043 know how you want to move forward with that then we can get some examples and designs for whatever you guys decide.

**Comment by dmugtasimov** [comment]
@buckyroberts "but I figured this would be one of the last bits of logic we implement once everything else is in place so we can wait on that - would like @dmugtasimov 's feedback on the best way to implement that as well." - not at all, since we already implemented the account root file and it is a part of the logic, and also because we need to have the very first "genesis block" (is it still a good name? should we use "blockchain snapshot" term instead? because having multiple genesis blocks sounds odd). So we need to implement it all together

**Comment by dmugtasimov** [comment]
"Another thing that we need to do is add all the fields and field types to each of the blocks here https://thenewboston.com/guide/block-types I am not sure how we want to display those but saw @dmugtasimov and @led76marx talking about a template engine which would be perfect for this. If you could get together and let @ma2b0043 know how you want to move forward with that then we can get some examples and designs for whatever you guys decide."

Let's keep it as is now. Once we implement https://github.com/thenewboston-developers/thenewboston-node/issues/95 we will figure out how to copy / proxy this information to the website.

The template engine is another story though.

**Comment by dmugtasimov** [comment]
@buckyroberts I will be very pydantic about the docs now if you do not mind :)

**Comment by dmugtasimov** [comment]
Guide
- Introduction
- Glossary
- Blocks
- Block Structure
- Block Types
- Nodes
- Flows
- Fork Prevention
- Scheduling
- Schedule Adjustments
- Accounts
- Account Lock
- Node Identifier
- Fees
- Best Practices
- Root Account File
- Blockchain Architecture

Governance
- Overview
- Usernames & Votes
- Locked Coins & Boosting
- Points & Refill Logic
- Coins vs. Points
- Election Process
- Budgets
- Rates & Amounts

**Comment by dmugtasimov** [comment]
@buckyroberts I will add incontext comments to each page. Please, see the links in the previous comment

**Comment by dmugtasimov** [comment]
done with introduction

**Comment by dmugtasimov** [comment]
@buckyroberts I partially reviewed Glosary, please, see comments in Google Doc. But I see that some of my comments from https://docs.google.com/document/d/1EgS2xZTVOSG_YiNWjXV6a1FSXErdQFcyOhmCstjI22A/edit# are not address. Instead of copying / rewording them in the new document I suggest to work in the original one replying to comments until we come to an agreement and I can mark a comment as resolved (I prefer to restrict marking as resolved by the comment author. This way we can make sure that the author satisfied with the resolution and not comment got missed).

I pause working on this task until getting a feedback. Maybe we need to have call to discuss. Please, let me know.

**Comment by ma2b0043** [comment]
> Guide
> 
> * Introduction -> https://docs.google.com/document/d/1bF1x-JwNCWcsaD3JImkiOZpoZh5dA8coXXuBlb-pfSA/edit?usp=sharing
> * Glossary -> https://docs.google.com/document/d/1D6YaCSr2oWj7Z_Wvb-e3vZQbIhX59mRWCIH55QFmxDw/edit?usp=sharing
> * Blocks
> * Block Structure
> * Block Types
> * Nodes
> * Flows
> * Fork Prevention
> * Scheduling
> * Schedule Adjustments
> * Accounts
> * Account Lock
> * Node Identifier
> * Fees
> * Best Practices
> * Root Account File
> * Blockchain Architecture
> 
> Governance
> 
> * Overview
> * Usernames & Votes
> * Locked Coins & Boosting
> * Points & Refill Logic
> * Coins vs. Points
> * Election Process
> * Budgets
> * Rates & Amounts

Hi @dmugtasimov can you allow me and @led76marx to edit these docs as well, that way we can all post our comments in each of document, per subsection? then @buckyroberts can view these docs one by one.

**Comment by led76marx** [comment]
@dmugtasimov Unless you and @buckyroberts have discussed otherwise, it would be beneficial to all of us if you keep working on commenting this content, if you have capacity of course. I think you enable Bucky, and you definitely help me and @ma2b0043 understand some parts of the network. No need to comment same things twice if context is repeated (plz mention, however, "see comment about x and y" in instances of an issue, if it is not obvious that is). If this helps our task, here are my comments for Blocks, Block Structure, Block Types, Nodes. Feel free to add yours in there. https://docs.google.com/document/d/16TDuVtkGUHNudKr43JkcXBscLD9kfNj_eAFKTZtokck/edit?usp=sharing Also, I will do an editorial pass at the end when content becomes more solid, no need to worry about that now.

**Comment by led76marx** [comment]
Added my comments about the rest of the Guide docs. Here is the full list of gdocs used for the Guides. (First two are Dmitry's).

- Introduction -> https://docs.google.com/document/d/1bF1x-JwNCWcsaD3JImkiOZpoZh5dA8coXXuBlb-pfSA/edit?usp=sharing
- Glossary -> https://docs.google.com/document/d/1D6YaCSr2oWj7Z_Wvb-e3vZQbIhX59mRWCIH55QFmxDw/edit?usp=sharing
- Blocks, Block Structure, Block Types, Nodes: https://docs.google.com/document/d/16TDuVtkGUHNudKr43JkcXBscLD9kfNj_eAFKTZtokck/edit?usp=sharing
- Flows till the end of Guides doc set: https://docs.google.com/document/d/1JGKs9k53WEf6qCXwnRZ29gnZEmH_Kj0hE7ucIAdDTR8/edit#

**Comment by led76marx** [comment]
Government docs:
https://docs.google.com/document/d/1A5v6RKRI4KK2_wifNsRw1ltUHfdFgt9kW-Us3A63pbU/edit#

**Comment by buckyroberts** [comment]
Awesome review guys! I will go through tomorrow and start on those updates :-)

**Comment by dmugtasimov** [comment]
@buckyroberts let's agree that the comments in the docs get resolved only after the changes appear on the website. This way we can make sure that every comment is addressed.

**Comment by dmugtasimov** [comment]
@buckyroberts meanwhile I will produce the description of new collaboration workflow, and maybe we will switch to it

**Comment by dmugtasimov** [comment]
This is a comment for myself to keep track on the documentation discussion.

**Comment by sno2** [comment]
Hey, I think there's a `account_number` query parameter on some of the API endpoints for getting transactions that were sent by a certain account used by the account manager.  Should that be included within this doc update?
